backend/tests/test_rate_limit_fix.py

```
@@ -0,0 +1,926 @@
+import uuid
+from datetime import datetime, timedelta, timezone
+from unittest.mock import Mock, MagicMock, patch
+from uuid import UUID
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, Session
+from sqlalchemy.pool import StaticPool
+
+
+class MockTenantRateLimit:
+    """Mock TenantRateLimit object for testing"""
+    def __init__(self, tenant_id, limit_type, limit_value, current_usage=0, window_start=None, window_end=None):
```

Ruff (0.14.7)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)